api/src/main/java/com/nicknathanjustin/streamercontracts/users/UserDto.java

```diff
@@ -8,14 +9,19 @@
 
 @Data
 public class UserDto {
+    @JsonView(View.PublicUser.class)
     private final String displayName;
 
+    @JsonView(View.PublicUser.class)
     private final String type;
 
+    @JsonView(View.PublicUser.class)
     private final String broadcasterType;
 
+    @JsonView(View.PublicUser.class)
     private final String description;
 
+    @JsonView(View.PublicUser.class)
```

Comment:
My hunch is we should remove the fields we dont want to expose from the DTO (then we wouldnt need to worry about these annotations). then only add the other fields (alertChannelId, viewCount) when needed. So our API's would look like #4 or #3 in this post: https://www.baeldung.com/spring-mvc-model-model-map-model-view

```
UserDto userDto = //someUserDto
String alertChannelId = //someValue
Map<String, String> map = new HashMap<>();
map.add("userDto", userDto)
map.add("alertChannelId", alertChannelId)
```

Reply:
So maybe I am understanding this incorrectly, but this is changing what is being returned to the FE (so the FE would have to be changed). The json that would be generated from this is

{
  userDto: {
    // user properties
  },
  alertChannelId: 'fewafewfaefdsxsafes',
}

Now everything isn't nicely wrapped in the userDto object. The link you posted uses this "Model" object to combine the properties of the map together and return to the FE...we don't have that here. This approach seems more complicated to me.

Reply:
Here is the reference that talks about jsonViews:

https://www.baeldung.com/jackson-json-view-annotation